### ISSUE TITLE
[Issue]: UNIPC  sampler hangs with DirectML

### ISSUE THREAD

**Issue by Draxredd** [vladmandic/automatic#1768]
### Issue Description

Using unPC sampler hangs on first pass after 2/3 its when on DirectML 
radeon RX7900XTX (24gb vram) and ryzen X7700 with 35gb ram

### Version Platform Description

 Starting SD.Next
11:28:47-195206 INFO     Python 3.10.11 on Windows
11:28:47-334311 INFO     Version: e002652a Wed Jul 19 15:09:01 2023 -0400
11:28:47-967131 INFO     Using DirectML Backend

### Acknowledgements

- [X] I have read the above and searched for existing issues

**Comment by Disty0** [comment]
Sounds like the same problem i had with IPEX & Intel ARC. 
The GPU itself was crashing without any logs in my case.

Solved it by running `torch.linalg.solve` on the CPU instead of the GPU:
```
#UniPC:
CondFunc('torch.linalg.solve',
    lambda orig_func, *args, **kwargs: orig_func(args[0].to("cpu"), args[1].to("cpu")).to(get_cuda_device_string()),
    lambda *args, **kwargs: args[1].device != torch.device("cpu"))
```

**Comment by TeutonJon78** [comment]
I just started running into this as well, and it's only on the diffusers backend. UniPC runs fine on the original backend with DirectML

**Comment by lshqqytiger** [comment]
Should be fixed in 7017a4a2a9170c37f3fb4b0b401c180d95a46881
